vultr/provider.go

```diff
@@ -0,0 +1,92 @@
+package vultr
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/libdns/libdns"
+)
+
+// Provider implements the libdns interfaces for Vultr
+// Adapted from libdns/digitalocean to work with the Vultr API
+type Provider struct {
+	Client
```

Comment:
Ah, maybe find a way to unexport this, since there's nothing the user can configure on this.